[Cloud Security] 198669 add support for multiple CVEs and improve vulnerability data grid, flyout and contextual flyout UI

alexreal1314 opened the pull request:
## Summary

Purpose of this PR is to update UI components which display vulnerabilities in order to support 3rd party integrations and in particularly Qualys, changes include:

- New column in Default view in table- "Vulnerability Title"
- Renaming "Vulnerability" column to "CVE ID"
- Introducing "Title" (Vulnerability Title) and "Subtitle" (CVE ID) in vulnerability flyout while also supporting multiple CVEs.
- Update severity handling to be case insensitive across relevant components - data grid, flyout, contextual flyout and insights tab data grid.


### Checklist

**Vulnerabilities data grid and Flyout**

- [x] Add new column with header 'Vulnerability Title' and the underlying field should be 'vulnerability.title' field.
- [x] change existing column with header 'Vulnerability' to 'CVE ID' and underlying field should be 'vulnerability.id'. 
- [x] aggregation query when grouping by CVE should fetch severities to be case-insensitive - since 3rd party integrations can introduce severities not in UPPER CASE.
- [x] Components related to vulnerability severity handle supported severity levels disregarding the case - vulnerabilities flyout and contextual flyout, data grid and insights tab data table.
- [x] Multiple CVEs under `vulnerability.id` is a supported case in Vulnerability Findings in Kibana.
- [x] missing `vulnerability.id` is handled gracefully in the UI - data grid
- [x] missing `vulnerability.id` is handled gracefully in the UI - flyout

**Insights tab data grid and contextual Flyout**

- [x] Add new column with header 'Vulnerability Title' and the underlying field should be 'vulnerability.title' field.
- [x] change existing column with header 'Vulnerability' to 'CVE ID' and underlying field should be 'vulnerability.id'. 
- [x] fix redirect link when clicking on a vulnerabilty row that redirect to the vulnerabilities tab - make sure search params and filters are working with vulnerability.id array of strings.
- [x] contextual flyout should support severities maps to be case insensitive.
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [x] [Flaky Test Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was used on any tests changed
- [x] The PR  description includes the appropriate Release Notes section, and the correct `release_note:*` label is applied per the [guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

**The following topics will be merged as part of another PR**
### Vulnerabilities data grid and Flyout
- arrays in package-related vulnerability fields are rendered correctly in the flyout
- arrays in package-related vulnerability fields are rendered correctly in the data grid.

### Insights tab data grid and contextual Flyout
- arrays in package-related vulnerability fields are rendered correctly in the contextual flyout
- arrays in package-related vulnerability fields are rendered correctly in the insights tab data grid.
**Vulnerabilities data grid grouping**

### Grouping by multi-value fields
- grouping by CVE should be handled properly in the UI.
- each field which might have multiple values - package.name, package.fixed_version and package.version should be flattened by each value separately so user have the max value and can group by each value.

### Identify risks
The UI changes in this PR address new integrations which has different structure to vulnerability fields which might cause UI bugs. Tests were modified and added to mitigate the risk of encountering issues.

### Screenshots


![Screenshot 2025-03-10 at 16 04 25](https://github.com/user-attachments/assets/8da0974e-dfb3-4aa7-a40f-daaeae53d8f5)

![Screenshot 2025-03-10 at 16 07 04](https://github.com/user-attachments/assets/5d079075-3774-46a1-9098-6e437fbd8e1e)

![Screenshot 2025-03-10 at 16 07 32](https://github.com/user-attachments/assets/5f68bebc-f532-4647-9671-40f41c4b039b)

![Screenshot 2025-03-10 at 16 04 40](https://github.com/user-attachments/assets/8bcba728-f513-434c-b764-5995737216d7)


Grouping before fix by multiple values
![image](https://github.com/user-attachments/assets/21756172-819c-4836-ba3a-79ae9ed6cbad)

Flatten by each value
![image](https://github.com/user-attachments/assets/d329d3f7-b499-4abb-8e40-6c8580be9202)

alexreal1314 committed:
add handling of missing vulnerability.id in the data grid and fix flyout not opening

alexreal1314 committed:
add handling of severity to be case insensitive in vulnerabilities datagrid and flyout

alexreal1314 committed:
update vulnerabilities data grid - add title column and rename Vulnerability to CVE ID column header

alexreal1314 committed:
add support for multiple cve in the flyout header

update flyout overview tab to support multiple cves, published date move to be above alerts section

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/280607)
* Commit: b950865e41d72edc1a0e33d310e5da6a62ff9808

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/280607#01956047-bea3-49d8-8959-37d0d32c20d9)

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280607","number":280607,"commit":"b950865e41d72edc1a0e33d310e5da6a62ff9808"}],"number":280607}
buildkite-pr-comment-->

alexreal1314 committed:
fix i18n translation files and create util function

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/280668)
* Commit: 335c02fd0d8b1540a7ca2f7b477b17fb1317d6bb

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/280668#019560c5-feaf-4717-b376-f952fde53f10)

### History
* :broken_heart: [Build #280607](https://buildkite.com/elastic/kibana-pull-request/builds/280607) failed b950865e41d72edc1a0e33d310e5da6a62ff9808

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280668","number":280668,"commit":"335c02fd0d8b1540a7ca2f7b477b17fb1317d6bb"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280607","number":280607,"commit":"b950865e41d72edc1a0e33d310e5da6a62ff9808"}],"number":280668}
buildkite-pr-comment-->

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/280722)
* Commit: 55981b354ee6e3c314ea60bdfaf619f7724f90f0

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/280722#0195612a-95b9-4fcc-a135-695d1314a2e7)

### History
* :broken_heart: [Build #280668](https://buildkite.com/elastic/kibana-pull-request/builds/280668) failed 335c02fd0d8b1540a7ca2f7b477b17fb1317d6bb
* :broken_heart: [Build #280607](https://buildkite.com/elastic/kibana-pull-request/builds/280607) failed b950865e41d72edc1a0e33d310e5da6a62ff9808

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280722","number":280722,"commit":"55981b354ee6e3c314ea60bdfaf619f7724f90f0"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280668","number":280668,"commit":"335c02fd0d8b1540a7ca2f7b477b17fb1317d6bb"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280607","number":280607,"commit":"b950865e41d72edc1a0e33d310e5da6a62ff9808"}],"number":280722}
buildkite-pr-comment-->

alexreal1314 committed:
fix vulnerabilities serverity flyout display, add test ids and fix unit tests

alexreal1314 committed:
add normalizeSeverity function unit tests

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/280882)
* Commit: a97001c6c7079221d83fa3542bbb9a9bb4dca4c8

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/280882#019562a0-75d8-4fe4-a197-5be6a9cc815c)

### History
* :broken_heart: [Build #280722](https://buildkite.com/elastic/kibana-pull-request/builds/280722) failed 55981b354ee6e3c314ea60bdfaf619f7724f90f0
* :broken_heart: [Build #280668](https://buildkite.com/elastic/kibana-pull-request/builds/280668) failed 335c02fd0d8b1540a7ca2f7b477b17fb1317d6bb
* :broken_heart: [Build #280607](https://buildkite.com/elastic/kibana-pull-request/builds/280607) failed b950865e41d72edc1a0e33d310e5da6a62ff9808

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280882","number":280882,"commit":"a97001c6c7079221d83fa3542bbb9a9bb4dca4c8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280722","number":280722,"commit":"55981b354ee6e3c314ea60bdfaf619f7724f90f0"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280668","number":280668,"commit":"335c02fd0d8b1540a7ca2f7b477b17fb1317d6bb"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280607","number":280607,"commit":"b950865e41d72edc1a0e33d310e5da6a62ff9808"}],"number":280882}
buildkite-pr-comment-->

alexreal1314 committed:
fix check types errors

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/280965)
* Commit: a2640738f153b19bab9919ffad63472612c8ade8

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/280965#0195653e-49d1-48f2-8734-03261ca1f92b)

### History
* :broken_heart: [Build #280882](https://buildkite.com/elastic/kibana-pull-request/builds/280882) failed a97001c6c7079221d83fa3542bbb9a9bb4dca4c8
* :broken_heart: [Build #280722](https://buildkite.com/elastic/kibana-pull-request/builds/280722) failed 55981b354ee6e3c314ea60bdfaf619f7724f90f0
* :broken_heart: [Build #280668](https://buildkite.com/elastic/kibana-pull-request/builds/280668) failed 335c02fd0d8b1540a7ca2f7b477b17fb1317d6bb
* :broken_heart: [Build #280607](https://buildkite.com/elastic/kibana-pull-request/builds/280607) failed b950865e41d72edc1a0e33d310e5da6a62ff9808

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280965","number":280965,"commit":"a2640738f153b19bab9919ffad63472612c8ade8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280882","number":280882,"commit":"a97001c6c7079221d83fa3542bbb9a9bb4dca4c8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280722","number":280722,"commit":"55981b354ee6e3c314ea60bdfaf619f7724f90f0"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280668","number":280668,"commit":"335c02fd0d8b1540a7ca2f7b477b17fb1317d6bb"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280607","number":280607,"commit":"b950865e41d72edc1a0e33d310e5da6a62ff9808"}],"number":280965}
buildkite-pr-comment-->

alexreal1314 committed:
update contextual flyout severity mapping to handle inensitive case values

update insights tab vulnerabilities column names

fix redirect from insights tab to vulnerabilties findings page

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/281994)
* Commit: 51148970342d447233634362b1dcb9cf0be74ce3

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/281994#01957785-a29b-4ba3-9feb-0ac140a61b94)

### History
* :broken_heart: [Build #280965](https://buildkite.com/elastic/kibana-pull-request/builds/280965) failed a2640738f153b19bab9919ffad63472612c8ade8
* :broken_heart: [Build #280882](https://buildkite.com/elastic/kibana-pull-request/builds/280882) failed a97001c6c7079221d83fa3542bbb9a9bb4dca4c8
* :broken_heart: [Build #280722](https://buildkite.com/elastic/kibana-pull-request/builds/280722) failed 55981b354ee6e3c314ea60bdfaf619f7724f90f0
* :broken_heart: [Build #280668](https://buildkite.com/elastic/kibana-pull-request/builds/280668) failed 335c02fd0d8b1540a7ca2f7b477b17fb1317d6bb
* :broken_heart: [Build #280607](https://buildkite.com/elastic/kibana-pull-request/builds/280607) failed b950865e41d72edc1a0e33d310e5da6a62ff9808

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/281994","number":281994,"commit":"51148970342d447233634362b1dcb9cf0be74ce3"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280965","number":280965,"commit":"a2640738f153b19bab9919ffad63472612c8ade8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280882","number":280882,"commit":"a97001c6c7079221d83fa3542bbb9a9bb4dca4c8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280722","number":280722,"commit":"55981b354ee6e3c314ea60bdfaf619f7724f90f0"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280668","number":280668,"commit":"335c02fd0d8b1540a7ca2f7b477b17fb1317d6bb"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280607","number":280607,"commit":"b950865e41d72edc1a0e33d310e5da6a62ff9808"}],"number":281994}
buildkite-pr-comment-->

alexreal1314 committed:
fix conflicts and update insights tab vulnerabilities table id and title cell render

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/281998)
* Commit: fcc5d865be93d5317161a04bead1e0da28fb111a

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/281998#019579ef-c9b9-4a8b-86d8-dcfdad0674e1)

### History
* :broken_heart: [Build #281994](https://buildkite.com/elastic/kibana-pull-request/builds/281994) failed 51148970342d447233634362b1dcb9cf0be74ce3
* :broken_heart: [Build #280965](https://buildkite.com/elastic/kibana-pull-request/builds/280965) failed a2640738f153b19bab9919ffad63472612c8ade8
* :broken_heart: [Build #280882](https://buildkite.com/elastic/kibana-pull-request/builds/280882) failed a97001c6c7079221d83fa3542bbb9a9bb4dca4c8
* :broken_heart: [Build #280722](https://buildkite.com/elastic/kibana-pull-request/builds/280722) failed 55981b354ee6e3c314ea60bdfaf619f7724f90f0
* :broken_heart: [Build #280668](https://buildkite.com/elastic/kibana-pull-request/builds/280668) failed 335c02fd0d8b1540a7ca2f7b477b17fb1317d6bb

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/281998","number":281998,"commit":"fcc5d865be93d5317161a04bead1e0da28fb111a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/281994","number":281994,"commit":"51148970342d447233634362b1dcb9cf0be74ce3"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280965","number":280965,"commit":"a2640738f153b19bab9919ffad63472612c8ade8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280882","number":280882,"commit":"a97001c6c7079221d83fa3542bbb9a9bb4dca4c8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280722","number":280722,"commit":"55981b354ee6e3c314ea60bdfaf619f7724f90f0"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280668","number":280668,"commit":"335c02fd0d8b1540a7ca2f7b477b17fb1317d6bb"}],"number":281998}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/281999)
* Commit: adf8c30a3647a39bd798353454f0812e18c8d511

### Failed CI Steps
* [Osquery Cypress Tests on Serverless #6](https://buildkite.com/elastic/kibana-pull-request/builds/281999#01957a01-1869-4f98-a50c-bd1850f430f5)

### History
* :broken_heart: [Build #281998](https://buildkite.com/elastic/kibana-pull-request/builds/281998) failed fcc5d865be93d5317161a04bead1e0da28fb111a
* :broken_heart: [Build #281994](https://buildkite.com/elastic/kibana-pull-request/builds/281994) failed 51148970342d447233634362b1dcb9cf0be74ce3
* :broken_heart: [Build #280965](https://buildkite.com/elastic/kibana-pull-request/builds/280965) failed a2640738f153b19bab9919ffad63472612c8ade8
* :broken_heart: [Build #280882](https://buildkite.com/elastic/kibana-pull-request/builds/280882) failed a97001c6c7079221d83fa3542bbb9a9bb4dca4c8
* :broken_heart: [Build #280722](https://buildkite.com/elastic/kibana-pull-request/builds/280722) failed 55981b354ee6e3c314ea60bdfaf619f7724f90f0
* :broken_heart: [Build #280668](https://buildkite.com/elastic/kibana-pull-request/builds/280668) failed 335c02fd0d8b1540a7ca2f7b477b17fb1317d6bb

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/281999","number":281999,"commit":"adf8c30a3647a39bd798353454f0812e18c8d511"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/281998","number":281998,"commit":"fcc5d865be93d5317161a04bead1e0da28fb111a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/281994","number":281994,"commit":"51148970342d447233634362b1dcb9cf0be74ce3"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280965","number":280965,"commit":"a2640738f153b19bab9919ffad63472612c8ade8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280882","number":280882,"commit":"a97001c6c7079221d83fa3542bbb9a9bb4dca4c8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280722","number":280722,"commit":"55981b354ee6e3c314ea60bdfaf619f7724f90f0"}],"number":281999}
buildkite-pr-comment-->

kibanamachine commented:
<!-- kbn-message-context:cloud-deploy-job -->
PR Cloud deployment started at: https://buildkite.com/elastic/kibana-deploy-cloud-from-pr/builds/102

kibanamachine commented:
<!-- kbn-message-context:cloud-deploy-job -->
Cloud deployment initiated, see credentials at: https://buildkite.com/elastic/kibana-deploy-cloud-from-pr/builds/102

elasticmachine commented:
Pinging @elastic/kibana-cloud-security-posture (Team:Cloud Security)

JordanSh commented on x-pack/solutions/security/packages/kbn-cloud-security-posture/public/src/utils/get_normalized_severity.ts:
```diff
@@ -0,0 +1,21 @@
+import type { VulnSeverity } from '@kbn/cloud-security-posture-common';
+import { VALID_SEVERITIES } from '@kbn/cloud-security-posture-common';
+
+const isVulnSeverity = (severity: string): severity is VulnSeverity =>
+  VALID_SEVERITIES.includes(severity as VulnSeverity);
+
+export const getNormalizedSeverity = (severity?: string): VulnSeverity => {
+  if (!severity) {
+    return 'UNKNOWN';
+  }
+
+  return (
+    isVulnSeverity(severity.toUpperCase()) ? severity.toUpperCase() : 'UNKNOWN'
+  ) as VulnSeverity;
+};
```
1. I think you can remove the casting by doing this:
```typescript
const isVulnSeverity = (severity: string): severity is VulnSeverity => 
     VALID_SEVERITIES.some(validSeverity => validSeverity === severity);
```

2. regarding
```
  if (!severity) {
    return 'UNKNOWN';
  }
```
are we sure we want to flag missing data is UKNOWN? maybe it should be left blank?

3. the casting at the end pretty much negates the entire process that we are doing, I'm suggesting some changes:
```typescript
  if (isVulnSeverity(severity)) {
    return severity; // Return original if already a valid severity
  }

  const upperCaseSeverity = severity.toUpperCase(); // now we only uppercasing in case the first check hasn't passed. plus putting this value in a constant will let us run the type guard check and remove the need of type casting

  if (isVulnSeverity(upperCaseSeverity)) { 
    return upperCaseSeverity;
  }

  return 'UNKNOWN';
```

4. regarding the `return 'UNKNOWN';` at the end. since we meant to be agnostic to the data, maybe we want to support 3P types of severities? changing customer's severity types to something else might cause them problems. should verify with product. 
 
@smriti0321 @tinnytintin10 @nick-alayil, can one of you please confirm point 2 and 4?

smriti0321 replied:
@JordanSh For #2 Yes "Unknown" severity is recognised category in vulnerability detection space. We should ideally show Severity- Unknown and not leave it blank like other fields. For #4 I am not very clear about ask, overall we should not assign "UNKNOWN" severity on our side if source is not providing this value.

JordanSh replied:
Thanks @smriti0321!
So just to summarize
We only assign severity UNKNOWN if the actual value is indeed "UNKNOWN", or if the severity value is `undefined`. For any other case we either assign our default values or any other value that the 3P has saved in this field.

nick-alayil replied:
**TL;DR:** We should enumerate 3P data as-is. We can make UI/UX adjustments for proper display, but no data transformations.

Thanks for the discussion everyone. Let me clarify a few points:

> are we sure we want to flag missing data is UKNOWN? maybe it should be left blank?

I agree we should leave it blank. Here's why:

1. Our primary goal is to enumerate 3P vuln findings data without enrichment at this stage. Making enrichment decisions now could create inconsistencies with future 3P data sources.
2. `UNKNOWN` is actually a legitimate severity value that some tools output explicitly. We shouldn't override or assign it when the original source doesn't provide this value.

> changing customer's severity types to something else might cause them problems.

Completely agree here. 👍

It's worth noting this code was originally built for CNVM specific use cases, where we made certain adjustments to handle missing data for better table presentation. Now that we're expanding to enumerate all 3P vuln data in the findings page, we should maintain the original 3P data integrity.

JordanSh replied:
@alexreal1314 since we will need to support empty severity and other random values, we will also need to address it in the UI. we will need to verify we can still group search and display it in the flyout

alexreal1314 replied:
@nick-alayil @smriti0321 @JordanSh thanks for the inputs.
just to make sure with some examples regarding severity:
1. low/medium/high/critical/unknown -> in any case we convert to upper case so we can use our colors for marking severities.
2. missing severity field/any other value -> stays the same.
![image](https://github.com/user-attachments/assets/cb22e304-be93-4837-a09b-77fc6afc554e)

nick-alayil replied:
> * low/medium/high/critical/unknown -> in any case we convert to upper case so we can use our colors for marking severities.

Yes, that sounds good 👍 

> * missing severity field/any other value -> stays the same.

Correct.

> do we want to display it like that in the data grid?

Yes please. 

> also need to think what to show in the flyout.

If there is no severity, we don't need to display anything as shown in your 2nd flyout screenshot.

JordanSh commented on x-pack/solutions/security/packages/kbn-cloud-security-posture/public/src/utils/query_utils.ts:
```diff
@@ -72,12 +72,24 @@ export const composeQueryFilters = (
-    createFilter(key, filterValue, dataViewId)
-  );
+  return Object.entries(filterParams).flatMap(([key, filterValue]) => {
+    if (Array.isArray(filterValue)) {
+      return filterValue.map((value) => createFilter(key, value, dataViewId));
+    }
+
+    if (!filterValue) {
+      return [];
+    }
+
+    return createFilter(key, filterValue, dataViewId);
+  });
 };
 
-export const createFilter = (key: string, filterValue: FilterValue, dataViewId: string): Filter => {
+export const createFilter = (
+  key: string,
+  filterValue: Exclude<FilterValue, string[]>,
+  dataViewId: string
+): Filter => {
```
can we add tests for those 2 functions after the changes?

JordanSh commented on x-pack/solutions/security/plugins/cloud_security_posture/public/components/cloud_security_data_table/cloud_security_data_table.tsx:
```diff
@@ -161,6 +161,25 @@ export const CloudSecurityDataTable = ({
   }, [persistedSettings, columnHeaders]);
 
+  // This is a mapper that adds vulnerability.id field as an empty string - needed for the flyout to open
+  const rowsWithVulnerabilityId = useMemo(
```
can this be done on the flyout level instead? i dont think we should run over all the items in the table. we can just have a default value for the id when the flyout is opening

alexreal1314 replied:
@JordanSh the issue is that the flyout doesn't open at all if 'vulnerability.id' key is missing.

JordanSh replied:
thats what im suggesting to change. lets make the flyout be able to open even without vuln id. thats what happening anyway. we just need to remove this condition from the flyout rather than add empty ids to all items (which is also data manipulation which we want to avoid as much as possible)

alexreal1314 replied:
@JordanSh in that case what should be the check instead of this one?
`const isCspVulnerabilityFinding = (
  source: Record<string, any> | undefined
): source is CspVulnerabilityFinding => {
  return source?.vulnerability?.id !== undefined;
};`

do we want some kind of other check maybe based on vendor? or we allow to open the flyout from any type of vulnerability?

alexreal1314 replied:
waiting for a decision on how should we check if we should open the vulnerability flyout - based on observer.vendor field or another field.

JordanSh reviewed (commented):
added some comments for now, still need to continue the review but i think some changes you can already start working on. I'm also adding paulo and max as reviewers since paulo is most the experienced with the table and max has the most knowledge with 3Ps

alexreal1314 committed:
add unit tests to composeQueryFilters and normalizeSeverity utility functions

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/284391)
* Commit: 9094c1b55effa4e70f6e16765db8fcdbc14c53d3

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/284391#01959ee4-aa78-4034-8201-11ea79a45eb4)

### History
* :yellow_heart: [Build #281999](https://buildkite.com/elastic/kibana-pull-request/builds/281999) was flaky adf8c30a3647a39bd798353454f0812e18c8d511
* :broken_heart: [Build #281998](https://buildkite.com/elastic/kibana-pull-request/builds/281998) failed fcc5d865be93d5317161a04bead1e0da28fb111a
* :broken_heart: [Build #281994](https://buildkite.com/elastic/kibana-pull-request/builds/281994) failed 51148970342d447233634362b1dcb9cf0be74ce3
* :broken_heart: [Build #280965](https://buildkite.com/elastic/kibana-pull-request/builds/280965) failed a2640738f153b19bab9919ffad63472612c8ade8
* :broken_heart: [Build #280882](https://buildkite.com/elastic/kibana-pull-request/builds/280882) failed a97001c6c7079221d83fa3542bbb9a9bb4dca4c8

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/284391","number":284391,"commit":"9094c1b55effa4e70f6e16765db8fcdbc14c53d3"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/281999","number":281999,"commit":"adf8c30a3647a39bd798353454f0812e18c8d511"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/281998","number":281998,"commit":"fcc5d865be93d5317161a04bead1e0da28fb111a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/281994","number":281994,"commit":"51148970342d447233634362b1dcb9cf0be74ce3"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280965","number":280965,"commit":"a2640738f153b19bab9919ffad63472612c8ade8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280882","number":280882,"commit":"a97001c6c7079221d83fa3542bbb9a9bb4dca4c8"}],"number":284391}
buildkite-pr-comment-->

seanrathier reviewed (commented):
Looks good! I left a few comments.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/284925)
* Commit: b575bfcdb8e922e6624d63414d966f05e76e105a

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/284925#0195a8da-19e0-4e56-9678-1ffbf40e08d5)

### History
* :broken_heart: [Build #284391](https://buildkite.com/elastic/kibana-pull-request/builds/284391) failed 9094c1b55effa4e70f6e16765db8fcdbc14c53d3
* :yellow_heart: [Build #281999](https://buildkite.com/elastic/kibana-pull-request/builds/281999) was flaky adf8c30a3647a39bd798353454f0812e18c8d511
* :broken_heart: [Build #281998](https://buildkite.com/elastic/kibana-pull-request/builds/281998) failed fcc5d865be93d5317161a04bead1e0da28fb111a
* :broken_heart: [Build #281994](https://buildkite.com/elastic/kibana-pull-request/builds/281994) failed 51148970342d447233634362b1dcb9cf0be74ce3
* :broken_heart: [Build #280965](https://buildkite.com/elastic/kibana-pull-request/builds/280965) failed a2640738f153b19bab9919ffad63472612c8ade8

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/284925","number":284925,"commit":"b575bfcdb8e922e6624d63414d966f05e76e105a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/284391","number":284391,"commit":"9094c1b55effa4e70f6e16765db8fcdbc14c53d3"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/281999","number":281999,"commit":"adf8c30a3647a39bd798353454f0812e18c8d511"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/281998","number":281998,"commit":"fcc5d865be93d5317161a04bead1e0da28fb111a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/281994","number":281994,"commit":"51148970342d447233634362b1dcb9cf0be74ce3"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280965","number":280965,"commit":"a2640738f153b19bab9919ffad63472612c8ade8"}],"number":284925}
buildkite-pr-comment-->

alexreal1314 commented on x-pack/solutions/security/plugins/cloud_security_posture/public/pages/vulnerabilities/utils/create_detection_rule_from_vulnerability.ts:
```diff
@@ -97,9 +100,25 @@ export const createDetectionRuleFromVulnerabilityFinding = async (
     );
   }
 
+  if (!vulnerabilityFinding.vulnerability.id) {
```
@seanrathier yes.

alexreal1314 commented on x-pack/solutions/security/packages/kbn-cloud-security-posture/public/src/utils/hooks_utils.ts:
changed to findings_query_builders since this is what this file contains.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/285021)
* Commit: 3f3bfc2e2db8800302d7de87c56fc1ea0f5e77f1
* [Cloud Deployment](https://kibana-pr-213039.kb.us-west2.gcp.elastic-cloud.com)

### History
* :green_heart: [Build #284989](https://buildkite.com/elastic/kibana-pull-request/builds/284989) succeeded 4e396a380f247fb62771e578b34f70f175204fe8
* :broken_heart: [Build #284925](https://buildkite.com/elastic/kibana-pull-request/builds/284925) failed b575bfcdb8e922e6624d63414d966f05e76e105a
* :broken_heart: [Build #284391](https://buildkite.com/elastic/kibana-pull-request/builds/284391) failed 9094c1b55effa4e70f6e16765db8fcdbc14c53d3
* :yellow_heart: [Build #281999](https://buildkite.com/elastic/kibana-pull-request/builds/281999) was flaky adf8c30a3647a39bd798353454f0812e18c8d511
* :broken_heart: [Build #281998](https://buildkite.com/elastic/kibana-pull-request/builds/281998) failed fcc5d865be93d5317161a04bead1e0da28fb111a
* :broken_heart: [Build #281994](https://buildkite.com/elastic/kibana-pull-request/builds/281994) failed 51148970342d447233634362b1dcb9cf0be74ce3

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/285021","number":285021,"commit":"3f3bfc2e2db8800302d7de87c56fc1ea0f5e77f1"},{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/284989","number":284989,"commit":"4e396a380f247fb62771e578b34f70f175204fe8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/284925","number":284925,"commit":"b575bfcdb8e922e6624d63414d966f05e76e105a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/284391","number":284391,"commit":"9094c1b55effa4e70f6e16765db8fcdbc14c53d3"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/281999","number":281999,"commit":"adf8c30a3647a39bd798353454f0812e18c8d511"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/281998","number":281998,"commit":"fcc5d865be93d5317161a04bead1e0da28fb111a"}],"number":285021}
buildkite-pr-comment-->

seanrathier reviewed (approved):
Approving my concerns,  thanks for making the changes. 👍

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/285552)
* Commit: 77f7e418484146ef65df599ed6b4e518f0ded9c7

### Failed CI Steps
* [Osquery Cypress Tests #8](https://buildkite.com/elastic/kibana-pull-request/builds/285552#0195afb9-22b5-4f4b-8a2f-c0e375d63127)

### History
* :green_heart: [Build #285021](https://buildkite.com/elastic/kibana-pull-request/builds/285021) succeeded 3f3bfc2e2db8800302d7de87c56fc1ea0f5e77f1
* :green_heart: [Build #284989](https://buildkite.com/elastic/kibana-pull-request/builds/284989) succeeded 4e396a380f247fb62771e578b34f70f175204fe8
* :broken_heart: [Build #284925](https://buildkite.com/elastic/kibana-pull-request/builds/284925) failed b575bfcdb8e922e6624d63414d966f05e76e105a
* :broken_heart: [Build #284391](https://buildkite.com/elastic/kibana-pull-request/builds/284391) failed 9094c1b55effa4e70f6e16765db8fcdbc14c53d3
* :yellow_heart: [Build #281999](https://buildkite.com/elastic/kibana-pull-request/builds/281999) was flaky adf8c30a3647a39bd798353454f0812e18c8d511

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/285552","number":285552,"commit":"77f7e418484146ef65df599ed6b4e518f0ded9c7"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/285021","number":285021,"commit":"3f3bfc2e2db8800302d7de87c56fc1ea0f5e77f1"},{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/284989","number":284989,"commit":"4e396a380f247fb62771e578b34f70f175204fe8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/284925","number":284925,"commit":"b575bfcdb8e922e6624d63414d966f05e76e105a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/284391","number":284391,"commit":"9094c1b55effa4e70f6e16765db8fcdbc14c53d3"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/281999","number":281999,"commit":"adf8c30a3647a39bd798353454f0812e18c8d511"}],"number":285552}
buildkite-pr-comment-->

opauloh commented on src/platform/packages/shared/kbn-grouping/src/components/grouping.tsx:
```diff
@@ -57,6 +57,11 @@ export interface GroupingProps<T> {
+  // usefull in combination with shouldFlattenMultiValueField param in GroupingQueryArgs
+  // because if the field is a multi-value field, and we emit each value separatly the size of the field will be ignored
+  // when filtering by it
+  shouldIgnoreFieldSize?: boolean;
```
I am not sure about exposing this prop as it is, it seems too specific as consumers would need prior knowledge of the implementation details in order to use it properly as the comments propose. Maybe it could be easier to understand an approach where the consumer passes an array of fields that should flatten multiple values and all the necessary logic is within the grouping package? (i.e calling shouldFlattenMultiValueField and passing the necessary argument to createGroupFilter to ignore the field size)

```typescript
flatennedMultiValueFields?: string[]
```

alexreal1314 replied:
good point, ill move this inner logic to the package.

alexreal1314 replied:
moved this logic to the package itself, also added some comment to warn about the casting of keys in the runtime_mappings since it affects negatively the performance of the query.

opauloh reviewed (commented):
Left some comments, overall great work @alexreal1314 💯

alexreal1314 commented on x-pack/solutions/security/plugins/cloud_security_posture/public/common/constants.ts:
```diff
@@ -203,6 +204,19 @@ export const VULNERABILITY_GROUPING_OPTIONS = {
+ * ECS schema unique field to describe the event
+ * https://www.elastic.co/guide/en/ecs/current/ecs-event.html
+ */
+export const EVENT_ID = 'event.id';
```
thanks Paulo, yes im on it, do we have access to the wiz docs? i would like to know which field is the most appropriate to  be mapped to event.id

maxcold replied:
discussed in Slack, I do have access to Wiz docs and we can use their `id` for the `event.id` .

alexreal1314 replied:
@opauloh created PRs both in the [docs](https://github.com/elastic/security-team/pull/12184) and [integrations](https://github.com/elastic/integrations/pull/13201) to make sure we have this field set.

alexreal1314 commented on x-pack/solutions/security/plugins/cloud_security_posture/public/common/constants.ts:
```diff
@@ -203,6 +204,19 @@ export const VULNERABILITY_GROUPING_OPTIONS = {
+
+export const VULNERABILITY_GROUPING_MULTIPLE_VALUE_FIELDS = [
+  VULNERABILITY_FIELDS.VULNERABILITY_ID,
+  'package.name',
+  'package.version',
+  'package.fixed_version',
```
@opauloh yes in qualys it might be an [array](https://www.elastic.co/guide/en/integrations/current/qualys_vmdr.html#:~:text=VMDR%22%0A%20%20%20%20%7D%2C%0A%20%20%20%20%22-,package,-%22%3A%20%7B).

I still didn't update the rendering of package fields, ill do it in a separate PR since im splitting the changes to few PR.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/285732)
* Commit: a1410321cd3db6fd43c270f3284bc4628cdbebf8
* [Cloud Deployment](https://kibana-pr-213039.kb.us-west2.gcp.elastic-cloud.com)

### History
* :broken_heart: [Build #285718](https://buildkite.com/elastic/kibana-pull-request/builds/285718) failed f5534aa33f669091c609dfefe5eeb4379a8876d1
* :yellow_heart: [Build #285552](https://buildkite.com/elastic/kibana-pull-request/builds/285552) was flaky 77f7e418484146ef65df599ed6b4e518f0ded9c7
* :green_heart: [Build #285021](https://buildkite.com/elastic/kibana-pull-request/builds/285021) succeeded 3f3bfc2e2db8800302d7de87c56fc1ea0f5e77f1
* :green_heart: [Build #284989](https://buildkite.com/elastic/kibana-pull-request/builds/284989) succeeded 4e396a380f247fb62771e578b34f70f175204fe8
* :broken_heart: [Build #284925](https://buildkite.com/elastic/kibana-pull-request/builds/284925) failed b575bfcdb8e922e6624d63414d966f05e76e105a
* :broken_heart: [Build #284391](https://buildkite.com/elastic/kibana-pull-request/builds/284391) failed 9094c1b55effa4e70f6e16765db8fcdbc14c53d3

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/285732","number":285732,"commit":"a1410321cd3db6fd43c270f3284bc4628cdbebf8"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/285718","number":285718,"commit":"f5534aa33f669091c609dfefe5eeb4379a8876d1"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/285552","number":285552,"commit":"77f7e418484146ef65df599ed6b4e518f0ded9c7"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/285021","number":285021,"commit":"3f3bfc2e2db8800302d7de87c56fc1ea0f5e77f1"},{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/284989","number":284989,"commit":"4e396a380f247fb62771e578b34f70f175204fe8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/284925","number":284925,"commit":"b575bfcdb8e922e6624d63414d966f05e76e105a"}],"number":285732}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/285827)
* Commit: 1577ac826239e5a18d9c6f4aa01efbf4b27addc5
* [Cloud Deployment](https://kibana-pr-213039.kb.us-west2.gcp.elastic-cloud.com)

### History
* :green_heart: [Build #285732](https://buildkite.com/elastic/kibana-pull-request/builds/285732) succeeded a1410321cd3db6fd43c270f3284bc4628cdbebf8
* :broken_heart: [Build #285718](https://buildkite.com/elastic/kibana-pull-request/builds/285718) failed f5534aa33f669091c609dfefe5eeb4379a8876d1
* :yellow_heart: [Build #285552](https://buildkite.com/elastic/kibana-pull-request/builds/285552) was flaky 77f7e418484146ef65df599ed6b4e518f0ded9c7
* :green_heart: [Build #285021](https://buildkite.com/elastic/kibana-pull-request/builds/285021) succeeded 3f3bfc2e2db8800302d7de87c56fc1ea0f5e77f1
* :green_heart: [Build #284989](https://buildkite.com/elastic/kibana-pull-request/builds/284989) succeeded 4e396a380f247fb62771e578b34f70f175204fe8

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/285827","number":285827,"commit":"1577ac826239e5a18d9c6f4aa01efbf4b27addc5"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/285732","number":285732,"commit":"a1410321cd3db6fd43c270f3284bc4628cdbebf8"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/285718","number":285718,"commit":"f5534aa33f669091c609dfefe5eeb4379a8876d1"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/285552","number":285552,"commit":"77f7e418484146ef65df599ed6b4e518f0ded9c7"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/285021","number":285021,"commit":"3f3bfc2e2db8800302d7de87c56fc1ea0f5e77f1"},{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/284989","number":284989,"commit":"4e396a380f247fb62771e578b34f70f175204fe8"}],"number":285827}
buildkite-pr-comment-->

opauloh commented on x-pack/solutions/security/plugins/cloud_security_posture/public/pages/vulnerabilities/vulnerabilities_finding_flyout/vulnerability_finding_flyout.tsx:
```diff
@@ -160,11 +165,80 @@ export const VulnerabilityFindingFlyout = ({
+    }
+
+    const renderVulnerabilityId = () => {
+      const id = Array.isArray(vulnerability.id) ? vulnerability.id[0] : vulnerability.id;
```
just an observation, but as we always render the first CVE below the title, it might highlight other CVEs in the title section than the one that was grouped, I found it a little confusing as I was expecting them to match, but not sure if that would also confuse users.

https://github.com/user-attachments/assets/d9e215ff-1ae9-4cf1-9bf1-0e8de4c3071c

opauloh reviewed (approved):
LGTM, added a few more comments but feel free to skip or go over it in a follow up PR

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/286354)
* Commit: 712fc14f204dc20c5f273d4b5ac7b45b1d7cec64
* [Cloud Deployment](https://kibana-pr-213039.kb.us-west2.gcp.elastic-cloud.com)

### History
* :green_heart: [Build #285827](https://buildkite.com/elastic/kibana-pull-request/builds/285827) succeeded 1577ac826239e5a18d9c6f4aa01efbf4b27addc5
* :green_heart: [Build #285732](https://buildkite.com/elastic/kibana-pull-request/builds/285732) succeeded a1410321cd3db6fd43c270f3284bc4628cdbebf8
* :broken_heart: [Build #285718](https://buildkite.com/elastic/kibana-pull-request/builds/285718) failed f5534aa33f669091c609dfefe5eeb4379a8876d1
* :yellow_heart: [Build #285552](https://buildkite.com/elastic/kibana-pull-request/builds/285552) was flaky 77f7e418484146ef65df599ed6b4e518f0ded9c7
* :green_heart: [Build #285021](https://buildkite.com/elastic/kibana-pull-request/builds/285021) succeeded 3f3bfc2e2db8800302d7de87c56fc1ea0f5e77f1
* :green_heart: [Build #284989](https://buildkite.com/elastic/kibana-pull-request/builds/284989) succeeded 4e396a380f247fb62771e578b34f70f175204fe8

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/286354","number":286354,"commit":"712fc14f204dc20c5f273d4b5ac7b45b1d7cec64"},{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/285827","number":285827,"commit":"1577ac826239e5a18d9c6f4aa01efbf4b27addc5"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/285732","number":285732,"commit":"a1410321cd3db6fd43c270f3284bc4628cdbebf8"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/285718","number":285718,"commit":"f5534aa33f669091c609dfefe5eeb4379a8876d1"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/285552","number":285552,"commit":"77f7e418484146ef65df599ed6b4e518f0ded9c7"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/285021","number":285021,"commit":"3f3bfc2e2db8800302d7de87c56fc1ea0f5e77f1"}],"number":286354}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/286893)
* Commit: 236cfa77cf8972cc43601069e9301ce0bbe5768f
* [Cloud Deployment](https://kibana-pr-213039.kb.us-west2.gcp.elastic-cloud.com)

### History
* :yellow_heart: [Build #286354](https://buildkite.com/elastic/kibana-pull-request/builds/286354) was flaky 712fc14f204dc20c5f273d4b5ac7b45b1d7cec64
* :green_heart: [Build #285827](https://buildkite.com/elastic/kibana-pull-request/builds/285827) succeeded 1577ac826239e5a18d9c6f4aa01efbf4b27addc5
* :green_heart: [Build #285732](https://buildkite.com/elastic/kibana-pull-request/builds/285732) succeeded a1410321cd3db6fd43c270f3284bc4628cdbebf8
* :broken_heart: [Build #285718](https://buildkite.com/elastic/kibana-pull-request/builds/285718) failed f5534aa33f669091c609dfefe5eeb4379a8876d1
* :yellow_heart: [Build #285552](https://buildkite.com/elastic/kibana-pull-request/builds/285552) was flaky 77f7e418484146ef65df599ed6b4e518f0ded9c7

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/286893","number":286893,"commit":"236cfa77cf8972cc43601069e9301ce0bbe5768f"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/286354","number":286354,"commit":"712fc14f204dc20c5f273d4b5ac7b45b1d7cec64"},{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/285827","number":285827,"commit":"1577ac826239e5a18d9c6f4aa01efbf4b27addc5"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/285732","number":285732,"commit":"a1410321cd3db6fd43c270f3284bc4628cdbebf8"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/285718","number":285718,"commit":"f5534aa33f669091c609dfefe5eeb4379a8876d1"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/285552","number":285552,"commit":"77f7e418484146ef65df599ed6b4e518f0ded9c7"}],"number":286893}
buildkite-pr-comment-->

alexreal1314 committed:
fix barrel imports and fix types of utility function used by SeverityStatusBadge

alexreal1314 committed:
fix finding matching refernce in case of multiple vulnerability.id values
